crates/stackable-operator/src/crd/git_sync/v1alpha1_impl.rs

```diff
@@ -0,0 +1,798 @@
+                "pipefail".to_string(),
+                "-c".to_string(),
+            ])
+            .args(vec![Self::create_git_sync_command(git_sync, one_time)])
+            .add_env_vars(env_vars.into())
+            .add_volume_mounts(volume_mounts.to_vec())
+            .context(AddVolumeMountSnafu)?
+            .resources(
+                ResourceRequirementsBuilder::new()
+                    .with_cpu_request("100m")
+                    .with_cpu_limit("200m")
+                    .with_memory_request("64Mi")
+                    .with_memory_limit("64Mi")
+                    .build(),
+            )
+            .build();
+        Ok(container)
+    }
+
+    fn create_git_sync_command(git_sync: &GitSync, one_time: bool) -> String {
+        let internal_args = [
+            Some(("--repo".to_string(), git_sync.repo.to_owned())),
+            Some(("--ref".to_string(), git_sync.branch.to_owned())),
+            Some(("--depth".to_string(), git_sync.depth.to_string())),
+            Some((
+                "--period".to_string(),
+                format!("{}s", git_sync.wait.as_secs()),
+            )),
+            Some(("--link".to_string(), GIT_SYNC_LINK.to_string())),
+            Some(("--root".to_string(), GIT_SYNC_ROOT_DIR.to_string())),
+            one_time.then_some(("--one-time".to_string(), "true".to_string())),
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<BTreeMap<_, _>>();
+
+        let internal_git_config = [(
+            GIT_SYNC_SAFE_DIR_OPTION.to_string(),
+            GIT_SYNC_ROOT_DIR.to_string(),
+        )]
+        .into_iter()
+        .collect::<BTreeMap<_, _>>();
+
+        let mut user_defined_args = BTreeMap::new();
+        // The key and value in Git configs are separated by a colon, but both
+        // can contain either escaped colons or unescaped colons if enclosed in
+        // quotes. To avoid parsing, just a vector is used instead of a map.
+        let mut user_defined_git_configs = Vec::new();
+
+        for (key, value) in &git_sync.git_sync_conf {
+            // The initial git-sync implementation in the airflow-operator
+            // (https://github.com/stackabletech/airflow-operator/pull/381)
+            // used this condition to find Git configs. It is also used here
+            // for backwards-compatibility:
+            if key.to_lowercase().ends_with("-git-config") {
+                // Roughly check if the user defined config contains an
+                // internally defined config and emit a warning in case.
+                if internal_git_config.keys().any(|key| value.contains(key)) {
+                    tracing::warn!("Config option {value:?} contains a value for {GIT_SYNC_SAFE_DIR_OPTION} that overrides
+                                the value of this operator. Git-sync functionality will probably not work as expected!");
+                }
+                user_defined_git_configs.push(value.to_owned());
```

Comment:
question: Do we want to allow that? This can potentially break what the operator wants to do. Basically user freedom vs opinionated approach.

Reply:
I think it is difficult to totally control/parse user input (as there are quite of lot of possible settings), so the approach I/we took was to only exclude (i.e. don't allow them to be overwritten) the few settings that we define and use internally.

Reply:
Alright, I'm happy to move forward as is. but I still think this might be worth a discussion down the line (going to `v1alpha2` might be a good time to re-iterate on a bunch of stuff).